### PR TITLE
list abandoned zones

### LINKED ISSUE #617: No way to view zone change history on deleted zone
**Describe the bug**
The only endpoint we expose for zone changes is `/zones/:id/changes`. This endpoint starts by checking if the zone exists and if it doesn't it returns a 404. This makes it impossible to view the change history of a deleted zone.

Not sure if it speeds up performance by checking if the zone is there before querying dynamo, so maybe keep doing that and only don't if the user is a support user?

**VinylDNS Version**
0.9.2-SNAPSHOT

### PR DESCRIPTION
**Pull request by Jay07GIT**
Fixes #617

Changes in this pull request:

- Added API for deleted zones history (zones/deleted/changes)
- Made Portal changes to view deleted zones history.
- Removed Cascade from zone_access table. We use this on comparing zone_change table .
- Added zone_status column in zone_access table to identify the deleted zones and used the same in deleting the group.

### CODE REVIEW

File: modules/mysql/src/main/scala/vinyldns/mysql/repository/MySqlZoneChangeRepository.scala [hunk 1]
```diff
@@ -110,6 +127,106 @@ class MySqlZoneChangeRepository
...
+
+          val zoneNotInZoneChange: List[ZoneChange] =
+            zoneChangeResults.filter(z=> !zoneResults.map(_.name).contains(z.zone.name) && z.zone.status != ZoneStatus.Active)
+
+          val deletedZoneResults: List[ZoneChange] =
+            zoneNotInZoneChange.filter(_.zone.status.equals(ZoneStatus.Deleted)).distinct.sortBy(_.zone.updated).reverse
+
+          val results: List[ZoneChange] =
+            if (zoneNameFilter.nonEmpty) {
+              deletedZoneResults.filter(r => r.zone.name.contains(zoneNameFilter.getOrElse("not found")))
```
Comment (Aravindh-Raju) [inline modules/mysql/src/main/scala/vinyldns/mysql/repository/MySqlZoneChangeRepository.scala:203]: I see zoneNameFilter and pagination are not covered by tests. We can add those tests in `MySqlZoneChangeRepositoryIntegrationSpec.scala` and increase the code coverage.

File: modules/mysql/src/main/scala/vinyldns/mysql/repository/MySqlZoneChangeRepository.scala [hunk 2]
```diff
@@ -110,6 +127,106 @@ class MySqlZoneChangeRepository
...
+            } else {
+              deletedZoneResults
+            }
+
+          val deletedZonesWithStartFrom: List[ZoneChange] = startFrom match {
```
Comment (Aravindh-Raju) [inline modules/mysql/src/main/scala/vinyldns/mysql/repository/MySqlZoneChangeRepository.scala:208]: Seems like pagination is not working as expected. Maybe a problem with how `startfrom` is used here. I have attached screenshots for reference. There were 4 items before pagination. But while paginating, I get the same results in every page. Needs to be fixed.

Screenshot 2023-08-02 at 12 34 36 PM
Screenshot 2023-08-02 at 12 37 41 PM
Screenshot 2023-08-02 at 12 37 46 PM

**Review by Aravindh-Raju — CHANGES_REQUESTED** [review]

File: modules/mysql/src/main/scala/vinyldns/mysql/repository/MySqlZoneChangeRepository.scala (hunk 2 above)
Comment (Jay07GIT) [reply modules/mysql/src/main/scala/vinyldns/mysql/repository/MySqlZoneChangeRepository.scala:208]: Startfrom changed from zone name to zone id. Now its working fine.

**Review by arpit4ever — APPROVED** [review]
LGTM
Tested (GUI & API) functionality locally.

**Review by nspadaccino — CHANGES_REQUESTED** [review]
I'm seeing a bug in the zones view.
If I log in as one user, create a group, and connect a zone, then log in with another user not in that group, I am able to see the zone in My Zones, and I am able to click on it. When I click on it, I get a 403 error (since that user doesn't have permission to view that private zone).
Screenshot 2023-08-16 at 11 19 09 AM
Other groups' private zones shouldn't appear in My Zones tab

**Comment by nspadaccino** [comment]
Also, it seems like wildcard search isn't working correctly for abandoned zones.
In My Zones, I can search for "o*" and see the zone "ok". But when I search for "o*" in abandoned zones, I get no results.
Screenshot 2023-08-16 at 11 30 24 AM
Screenshot 2023-08-16 at 11 30 32 AM
Screenshot 2023-08-16 at 11 30 40 AM
Screenshot 2023-08-16 at 11 30 47 AM

**Comment by Jay07GIT** [comment]
> Also, it seems like wildcard search isn't working correctly for abandoned zones. In My Zones, I can search for "o*" and see the zone "ok". But when I search for "o*" in abandoned zones, I get no results. Screenshot 2023-08-16 at 11 30 24 AM Screenshot 2023-08-16 at 11 30 32 AM Screenshot 2023-08-16 at 11 30 40 AM Screenshot 2023-08-16 at 11 30 47 AM

Yes. its right . Since we used scala filter we dont need wildcard for seach. we can jus give "o" to search ok. zones.

**Comment by Jay07GIT** [comment]
> I'm seeing a bug in the zones view. If I log in as one user, create a group, and connect a zone, then log in with another user not in that group, I am able to see the zone in My Zones, and I am able to click on it. When I click on it, I get a 403 error (since that user doesn't have permission to view that private zone). Screenshot 2023-08-16 at 11 19 09 AM Other groups' private zones shouldn't appear in My Zones tab

Oh yes. Updated the code. Now its working fine.

**Review by arpit4ever — APPROVED** [review]
LGTM

**Review by nspadaccino — APPROVED** [review]
looks good

**Pull request merged**